ksqldb-serde/src/main/java/io/confluent/ksql/serde/connect/ConnectDataTranslator.java

```diff
@@ -227,7 +227,7 @@ private Object toKsqlValue(
         }
       case INT32:
         if (schema.name() == Time.LOGICAL_NAME) {
-          return new java.sql.Time(((Number) convertedValue).longValue());
+          return new java.sql.Time(((Number) convertedValue).intValue());
```

Comment:
I assumed that being the number INT32, then `intValue` won't return an overflowed/underflowed value if it has a higher/lower number, right?

Reply:
Yeah, Connect does this check.